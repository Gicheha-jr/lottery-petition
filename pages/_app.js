import { appWithTranslation } from "next-i18next";
import nextI18NextConfig from "../next-i18next.config.js";
import React, { useState } from "react";
import "../styles/globals.css";
import { useTranslation } from "next-i18next";
import { useRouter } from "next/router";
import { wrapper } from "../store/store";
import { useSelector, useDispatch } from "react-redux";
import { todosLoading, fetchTodos } from "../store/layout/layoutSlice";

import {
  TeamOutlined,
  AimOutlined,
  InfoCircleOutlined,
  PhoneOutlined,
  TranslationOutlined,
  MenuFoldOutlined,
  MenuUnfoldOutlined,
} from "@ant-design/icons";
import { Layout, Menu, Button, Row, Col, theme } from "antd";

const { Header, Content, Sider, Footer } = Layout;

const App = ({ Component, ...rest }) => {
  const { props } = wrapper.useWrappedStore(rest);
  const { layout } = useSelector((state) => state);
  const dispatch = useDispatch();

  const {
    token: { colorBgContainer },
  } = theme.useToken();

  const { t } = useTranslation("common");
  const [current, setCurrent] = useState("1");
  const [collapsed, setCollapsed] = useState(true);

  const router = useRouter();

  const onToggleLanguageClick = (newLocale) => {
    const { pathname, asPath, query } = router;
    router.push({ pathname, query }, asPath, { locale: newLocale });
  };

  const items1 = [
    {
      key: "1",
      icon: <AimOutlined />,
      label: "Petition",
    },
    {
      key: "2",
      icon: <InfoCircleOutlined />,
      label: t("button_text"),
    },
    {
      key: "3",
      icon: <TeamOutlined />,
      label: "Sponsors & Supporters",
    },
    {
      key: "4",
      icon: <PhoneOutlined />,
      label: "Contacts" + " + " + layout.status,
    },
    {
      key: "5",
      icon: <TranslationOutlined />,
      label: "Translation",
      children: [
        {
          label: "English",
          key: "lang:en",
        },
        {
          label: "Swahili",
          key: "lang:sw",
        },
      ],
    },
  ];

  const onMenuClicked = (e) => {
    // console.log('click ', e);
    switch (e.key) {
      case "lang:en":
        onToggleLanguageClick("en");
        break;
      case "lang:sw":
        onToggleLanguageClick("sw");
        break;
      default:
        setCurrent(e.key);
        dispatch(fetchTodos());
        break;
    }
  };

  return (
    <Layout>
      <Header
        style={{
          position: "sticky",
          top: 0,
          zIndex: 1,
          padding: 0,
          paddingRight: 24,
          background: colorBgContainer,
        }}
      >
        <p className="logo">Lottery Petition</p>
        <Row style={{ minHeight: "100%" }}>
          <Col xs={0} sm={0} md={0} lg={24} xl={24}>
            <Menu
              onClick={onMenuClicked}
              style={{ justifyContent: "flex-end" }}
              mode="horizontal"
              selectedKeys={[current]}
              items={items1}
            />
          </Col>
          <Col xs={24} sm={24} md={24} lg={0} xl={0}>
            <Button
              type="secondary"
              onClick={() => setCollapsed(!collapsed)}
              className="trigger"
            >
              {collapsed ? <MenuFoldOutlined /> : <MenuUnfoldOutlined />}
            </Button>
          </Col>
        </Row>
      </Header>
      <Layout className="site-layout">
        <Content
          style={{
            margin: "24px 16px",
            padding: 24,
            minHeight: 280,
            background: colorBgContainer,
          }}
        >
          <Component {...props.pageProps} />
        </Content>
        <Sider
          trigger={null}
          collapsedWidth="0"
          collapsible
          collapsed={collapsed}
          style={{
            overflow: "auto",
            height: "100vh",
            position: "fixed",
            right: 0,
            top: 64,
            bottom: 0,
          }}
        >
          <Menu mode="inline" defaultSelectedKeys={[current]} items={items1} />
        </Sider>
      </Layout>
      <Footer style={{ textAlign: "center" }}>
        Kay Qube Design Â©2023 Created by Premar Systems Ltd
      </Footer>
    </Layout>
  );
};

export default wrapper.withRedux(appWithTranslation(App, nextI18NextConfig));
