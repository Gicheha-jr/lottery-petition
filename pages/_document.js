import { Html, Head, Main, NextScript } from 'next/document'
import i18nextConfig from '../next-i18next.config';

export default function Document({ props }) {
    console.log("Document props: ");
    console.log(props);

const currentLocale =
      props?.__NEXT_DATA__?.locale ??
      i18nextConfig.i18n.defaultLocale;

console.log("i18nextConfig.i18n.defaultLocale: " + i18nextConfig.i18n.defaultLocale);

  return (
    <Html lang={currentLocale}>
      <Head>
        <meta charSet="utf-8" />
      </Head>
      <body>
        <Main />
        <NextScript />
      </body>
    </Html>
  )
}