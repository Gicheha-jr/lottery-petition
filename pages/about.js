import Head from "next/head";
// import styles from "../styles/Home.module.css";
import Link from "next/link";
import { Button, Card, Col, Row } from "antd";
import { ShareAltOutlined, FacebookOutlined } from "@ant-design/icons";
import { withTranslation } from "next-i18next";
import { serverSideTranslations } from "next-i18next/serverSideTranslations";
const { Meta } = Card;

function Home({ t, i18n }) {
  return (
    <div>
      <Head>
        <title>About :: Petition & Ndirangu Gachunia</title>
        <link rel="icon" href="/favicon.ico" />
      </Head>

      <Row justify="center" align="stretch" gutter={[8, 16]}>
        <Col xs={24} sm={24} md={12} lg={8} xl={8}>
          <Card
            // hoverable
            style={{ height: "100%" }}
            actions={[
              <Button type="dashed" icon={<ShareAltOutlined />}>
                Share
              </Button>,
            ]}
            cover={
              <img
                alt="example"
                src="https://os.alipayobjects.com/rmsportal/QBnOOoLaAfKPirc.png"
              />
            }
          >
            <Meta
              title="Ndirangu Gachunia"
              description="CS Public Service, Nyeri County & CEO, CEREB : Sed euismod cursus nibh a condimentum. Nulla venenatis scelerisque massa. Aliquam justo orci, venenatis ut justo vitae, rutrum placerat risus. Praesent luctus eros vel tellus ultricies lobortis. Praesent dignissim urna commodo massa pellentesque pulvinar. Nullam commodo malesuada tristique."
            />
          </Card>
        </Col>
        <Col xs={24} sm={24} md={12} lg={8} xl={8}>
          <Card
            // hoverable
            style={{ height: "100%" }}
            actions={[
              <Button type="dashed" icon={<FacebookOutlined />}>
                See Facebook Post
              </Button>,
            ]}
          >
            <Meta
              title="CS Ndirangu Gachunia's Commitment"
              description={
                <p>
                  I listen to kikuyu Radio especially when driving. I like Sec
                  Gen and Board Chair coz l find them intelligent. But one
                  annoying thing that makes me switch off is the lottery. The
                  continuous advertising of people gambling to win some price is
                  driving our community in the wrong direction. I am the CEO
                  Central Region economic Bloc and I think about the Mountain
                  (ruriri) economically more often than not. We are the
                  entrepreneurial capital of the nation contributing 28% of the
                  GDP with a 10% population and owning 60% of Nairobi. Fact.
                  Lotteries are 'retrogressive taxes' on low income earners.
                  Hardly can you find successive people or men of real faith
                  playing lotteries. If on some dumb day I played and won I
                  would pay someone to pose on that photo. I can't live life
                  defined by randomness. We cannot feed our new generation that
                  success comes from LUCK and external factors. Successful
                  people according to Marshall Goldsmith believe that success
                  comes from sheer force of personality , talent, brainpower and
                  your inner drive. Given a chance i would bet on myself. This
                  gives me power to go to work and compete like a pro even when
                  everything else tells me not to. Am blind to negativity,
                  destiny crashers, excuses and naysayers. You put a block on my
                  path of progress and like a river i must find the strength to
                  navigate and keep walking like mutembei. Our media must
                  reflect our reality. They must stay true to our dreams and
                  aspirations as a community not feeding us garbage daily and
                  hoping for the best. They must account for every suicide,
                  broken home and all on the account of failed gambling. I find
                  it very insulting when it's linked to Faith in the God that I
                  believe in. Trail your trash as we say in the Circular Economy
                  . Account for your victims because they contribute to your
                  daily Harambees. Happy Holidays
                </p>
              }
            />
          </Card>
        </Col>
        <Col xs={24} sm={24} md={12} lg={8} xl={8}>
          <Card style={{ height: '100%' }}>
            <Meta
              title="Ndirangu Gachunia on Nyeri Mohoro NewsHub"
              description={
                <iframe
                  src="https://www.facebook.com/plugins/video.php?height=321&href=https%3A%2F%2Fwww.facebook.com%2Fnyerimohoro%2Fvideos%2F690811256087604%2F&show_text=true&width=560&t=0"
                  width="100%"
                  // height="1000"
                  style={{ border: "none", overflow: "hidden" }}
                  scrolling="no"
                  frameborder="0"
                  allowfullscreen="true"
                  allow="autoplay; clipboard-write; encrypted-media; picture-in-picture; web-share"
                  allowFullScreen="true"
                ></iframe>
              }
            />
          </Card>
        </Col>
      </Row>
    </div>
  );
}

export const getStaticProps = async ({ locale }) => ({
  props: {
    ...(await serverSideTranslations(locale ?? "en", ["common"])),
  },
});

export { Home };
export default withTranslation()(Home);
