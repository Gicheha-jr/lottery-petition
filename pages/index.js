import { useState, useRef } from "react";
import Head from "next/head";
import styles from "../styles/Home.module.css";

import { withTranslation } from "next-i18next";
import { serverSideTranslations } from "next-i18next/serverSideTranslations";
import SignPetition from "../components/sign-petition";
import { TeamOutlined } from "@ant-design/icons";
import { Button, Card, Col, Modal, Row } from "antd";
const { Meta } = Card;

function Home({ t, i18n }) {
  const [showModal, setShowModal] = useState(false);
  const submitButtonRef = useRef();

  return (
    <>
      <Head>
        <title>{t("title")}</title>
        <link rel="icon" href="/favicon.ico" />
      </Head>

      <Row>
        <Col
          xs={{ span: 24, offset: 0 }}
          sm={{ span: 24, offset: 0 }}
          md={{ span: 20, offset: 2 }}
          lg={{ span: 18, offset: 3 }}
          xl={{ span: 16, offset: 4 }}
        >
          <Card bordered={false}>
            <Meta
              avatar={<TeamOutlined />}
              title={
                <span style={{ whiteSpace: "normal" }}>
                  {t("petition_heading")}
                </span>
              }
              description={
                <>
                  <p>{t("petition")}</p>
                  <ul>
                    <li>{t("petition_1")}</li>
                    <li>{t("petition_2")}</li>
                  </ul>
                </>
              }
            />
          </Card>
        </Col>
      </Row>
      <br />
      <div className={styles.container}>
        <Button
          data-testid="sign-petition-button"
          size="large"
          type="primary"
          shape="round"
          onClick={() => setShowModal(true)}
        >
          {t("sign_btn_text")} &nbsp; &gt;
        </Button>
      </div>

      <Modal
        open={showModal}
        onOk={() => {
          submitButtonRef.current.dispatchEvent(new MouseEvent("click"));
        }}
        onCancel={() => setShowModal(false)}
        type="warning"
        cancelText={t("dismiss_modal_btn")}
        okText={t("sign_btn_text")}
      >
        <SignPetition
          {...{ t, i18n }}
          submitButtonRef={submitButtonRef}
        ></SignPetition>
      </Modal>
    </>
  );
}

export const getStaticProps = async ({ locale }) => ({
  props: {
    ...(await serverSideTranslations(locale ?? "en", ["index", "common"])),
  },
});

export { Home };
export default withTranslation(["index", "common"])(Home);
