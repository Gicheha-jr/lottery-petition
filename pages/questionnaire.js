import { useEffect, useState } from "react";
import Head from "next/head";
// import styles from "../styles/Contacts.module.css";

import { Button, Card, Col, Form, Input, Row } from "antd";
import { CommentOutlined } from "@ant-design/icons";

import { withTranslation } from "next-i18next";
import { serverSideTranslations } from "next-i18next/serverSideTranslations";

function Home({ t, i18n }) {
  const onFinish = (values) => {
    console.log("Success:", values);
  };

  const onFinishFailed = (errorInfo) => {
    console.log("Failed:", errorInfo);
  };

  const [form] = Form.useForm();
  const [currentLanguage, setCurrentLanguage] = useState(i18n.language);
  useEffect(() => {
    // language changed
    if (currentLanguage !== i18n.language) {
      setCurrentLanguage(i18n.language);

      ["name", "email", "phone", "message"].forEach((name) =>
        form.setFields([{ name: name, errors: "" }])
      );
    }
  });

  return (
    <>
      <Head>
        <title>{t("title")}</title>
        <link rel="icon" href="/favicon.ico" />
      </Head>

      <Row justify="center" align="stretch" gutter={[8, 16]}>
        <Col
          xs={{ span: 24, offset: 0 }}
          sm={{ span: 24, offset: 0 }}
          md={{ span: 20, offset: 2 }}
          lg={{ span: 18, offset: 3 }}
          xl={{ span: 16, offset: 4 }}
        >
          <Card
            title={
              <>
                <CommentOutlined /> &nbsp;
                <span style={{ whiteSpace: "normal" }}>{t("contact_us")}</span>
              </>
            }
            bordered={false}
          >
            <Form
              name="basic"
              labelCol={{ span: 8 }}
              wrapperCol={{ span: 16 }}
              style={{ maxWidth: 600 }}
              onFinish={onFinish}
              onFinishFailed={onFinishFailed}
              autoComplete="off"
              form={form}
            >
              <Form.Item
                label={t("full_name")}
                name="name"
                rules={[
                  {
                    required: true,
                    message: t("error_msg_prefix", { LABEL: t("full_name") }),
                  },
                ]}
              >
                <Input />
              </Form.Item>

              <Form.Item
                label={t("email")}
                name="email"
                rules={[
                  {
                    type: "email",
                    required: true,
                    message: t("error_msg_prefix", { LABEL: t("email") }),
                  },
                ]}
              >
                <Input />
              </Form.Item>

              <Form.Item
                label={t("phone_number")}
                name="phone"
                rules={[
                  {
                    required: true,
                    message: t("error_msg_prefix", {
                      LABEL: t("phone_number"),
                    }),
                  },
                ]}
              >
                <Input />
              </Form.Item>

              <Form.Item
                label={t("message")}
                name="message"
                rules={[
                  {
                    required: true,
                    message: t("error_msg_prefix", { LABEL: t("message") }),
                  },
                ]}
              >
                <Input.TextArea rows={6} />
              </Form.Item>

              <Form.Item wrapperCol={{ offset: 20, span: 16 }}>
                <Button type="primary" htmlType="submit">
                  {t("submit")}
                </Button>
              </Form.Item>
            </Form>
          </Card>
        </Col>
      </Row>
    </>
  );
}

export const getStaticProps = async ({ locale }) => ({
  props: {
    ...(await serverSideTranslations(locale ?? "en", ["contacts", "common"])),
  },
});

export { Home };
export default withTranslation(["contacts", "common"])(Home);
