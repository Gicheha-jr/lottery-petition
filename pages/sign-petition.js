import Head from "next/head";
// import styles from "../styles/Home.module.css";
import Link from "next/link";

import { withTranslation } from "next-i18next";
import { serverSideTranslations } from "next-i18next/serverSideTranslations";

function Home({ t, i18n }) {

  return (
    <>
      <Head>
        <title>Sign :: Lottery Petition</title>
        <link rel="icon" href="/favicon.ico" />
      </Head>
      
    </>
  );
}

export const getStaticProps = async ({ locale }) => ({
  props: {
    ...(await serverSideTranslations(locale ?? "en", ["common"])),
  },
});

export { Home };
export default withTranslation()(Home);
