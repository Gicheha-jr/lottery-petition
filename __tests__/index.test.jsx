import { render, screen } from "@testing-library/react";
import { Home } from "../pages/index";
import "@testing-library/jest-dom";

describe("Home", () => {
  
  it("renders homepage unchanged", () => {
    const { container } = render(<Home t={(key) => key} />);
    expect(container).toMatchSnapshot();
  });

  it("renders translate button", () => {
    render(<Home t={key => key} />);

    const heading = screen.getByTestId("translate-button");

    expect(heading).toBeInTheDocument();
  });
});
