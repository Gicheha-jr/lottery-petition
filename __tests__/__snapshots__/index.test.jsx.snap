// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders homepage unchanged 1`] = `
<div>
  <div
    class="jsx-25781291bf750 container"
  >
    <main
      class="jsx-25781291bf750"
    >
      <h1
        class="jsx-25781291bf750 title"
      >
        Read 
        <a
          href="/posts/first-post"
        >
          this page!
        </a>
      </h1>
      <button
        class="jsx-25781291bf750"
        data-testid="translate-button"
      >
        button_text
      </button>
      <p
        class="jsx-25781291bf750 description"
      >
        Get started by editing 
        <code
          class="jsx-25781291bf750"
        >
          pages/index.js
        </code>
      </p>
      <div
        class="jsx-25781291bf750 grid"
      >
        <a
          class="jsx-25781291bf750 card"
          href="https://nextjs.org/docs"
        >
          <h3
            class="jsx-25781291bf750"
          >
            Documentation →
          </h3>
          <p
            class="jsx-25781291bf750"
          >
            Find in-depth information about Next.js features and API.
          </p>
        </a>
        <a
          class="jsx-25781291bf750 card"
          href="https://nextjs.org/learn"
        >
          <h3
            class="jsx-25781291bf750"
          >
            Learn →
          </h3>
          <p
            class="jsx-25781291bf750"
          >
            Learn about Next.js in an interactive course with quizzes!
          </p>
        </a>
        <a
          class="jsx-25781291bf750 card"
          href="https://github.com/vercel/next.js/tree/master/examples"
        >
          <h3
            class="jsx-25781291bf750"
          >
            Examples →
          </h3>
          <p
            class="jsx-25781291bf750"
          >
            Discover and deploy boilerplate example Next.js projects.
          </p>
        </a>
        <a
          class="jsx-25781291bf750 card"
          href="https://vercel.com/import?filter=next.js&utm_source=create-next-app&utm_medium=default-template&utm_campaign=create-next-app"
        >
          <h3
            class="jsx-25781291bf750"
          >
            Deploy →
          </h3>
          <p
            class="jsx-25781291bf750"
          >
            Instantly deploy your Next.js site to a public URL with Vercel.
          </p>
        </a>
      </div>
    </main>
    <footer
      class="jsx-25781291bf750"
    >
      <a
        class="jsx-25781291bf750"
        href="https://vercel.com?utm_source=create-next-app&utm_medium=default-template&utm_campaign=create-next-app"
        rel="noopener noreferrer"
        target="_blank"
      >
        Powered by
         
      </a>
    </footer>
  </div>
</div>
`;
